utils/handles/signal.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "utils/handles/signal.h"
+
+#if !defined(_WIN32) && defined(__unix__) || defined(__unix) || \
+    (defined(__APPLE__) && defined(__MACH__))
+
+/*
+ * @para universal_signal_num_t SIG: assign the signal number which is going to
+ * handle.
+ * @para void (*signal_handler)(): signal handler of each signal needs
+ */
+static inline __sighandler_t register_signal(
+    universal_signal_num_t SIG, void (*register_signal_handler)()) {
+  return signal((int)SIG, register_signal_handler);
+}
+
+#elif defined(_WIN32)
+
+BOOL WINAPI signal_handler_WIN(DWORD dwCtrlType) {
+  switch (fdwCtrlType) {
+    case CTRL_C_EVENT:
+      // TODO get the funtion with map(ctrl_c)
+
+      log_info(logger_id, "Ctrl-C event\n\n");
+
+      return SIGNAL_SUCCESS;
+    default:
+      return SIGNAL_ERROR;
+  }
+}
+
+static inline __sighandler_t register_signal(
```

Comment:
static inline functions such as register_signal should appear in headers rather than C source.